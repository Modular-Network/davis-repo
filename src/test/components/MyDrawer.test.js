import React from 'react';
import { shallow } from 'enzyme';
import MyDrawer from 'components/MyDrawer';

describe('MyDrawer', () => {
    const drawer = shallow(<MyDrawer />);

    //console.log(drawer.debug()); 

    it('renders properly', () => {
        expect(drawer).toMatchSnapshot();
    });

    it('initializes the drawer `state`', () => {
        expect(drawer.state()).toEqual({ open: true });
    });

    it('contains the main drawer container', () => {
        expect(drawer.find('.drawer-container').exists()).toBe(true);
    });

    it('contains the mui Drawer component', () => {
        expect(drawer.find('.drawer-container').children('Drawer').length).toEqual(1);                
    });

    it('contains logo in Drawer component', () =>{
        expect(drawer.find('.logo').exists()).toBe(true);
    });

    it('contains route links in Drawer component', () =>{
        expect(drawer.find('Drawer').children('Link').length).toEqual(3);
    });

});