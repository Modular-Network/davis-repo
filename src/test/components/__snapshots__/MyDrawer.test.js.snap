// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyDrawer renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyDrawer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="drawer-container"
>
        <Drawer
                disableSwipeToOpen={false}
                docked={true}
                open={true}
                openSecondary={false}
                swipeAreaWidth={30}
                width={null}
                zDepth={2}
        >
                <div
                        className="logo"
                >
                        BLOSYM
                </div>
                <Link
                        className="nav-link"
                        replace={false}
                        to="/"
                >
                        <MenuItem
                                anchorOrigin={
                                        Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                                }
                                checked={false}
                                desktop={false}
                                disabled={false}
                                focusState="none"
                                insetChildren={false}
                                targetOrigin={
                                        Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                                }
                        >
                                Dashboard
                        </MenuItem>
                </Link>
                <Link
                        className="nav-link"
                        replace={false}
                        to="/wallets"
                >
                        <MenuItem
                                anchorOrigin={
                                        Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                                }
                                checked={false}
                                desktop={false}
                                disabled={false}
                                focusState="none"
                                insetChildren={false}
                                targetOrigin={
                                        Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                                }
                        >
                                Wallets
                        </MenuItem>
                </Link>
                <Link
                        className="nav-link"
                        replace={false}
                        to="/private"
                >
                        <MenuItem
                                anchorOrigin={
                                        Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                                }
                                checked={false}
                                desktop={false}
                                disabled={false}
                                focusState="none"
                                insetChildren={false}
                                targetOrigin={
                                        Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                                }
                        >
                                Private
                        </MenuItem>
                </Link>
        </Drawer>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Drawer
          disableSwipeToOpen={false}
          docked={true}
          open={true}
          openSecondary={false}
          swipeAreaWidth={30}
          width={null}
          zDepth={2}
>
          <div
                    className="logo"
          >
                    BLOSYM
          </div>
          <Link
                    className="nav-link"
                    replace={false}
                    to="/"
          >
                    <MenuItem
                              anchorOrigin={
                                        Object {
                                                  "horizontal": "right",
                                                  "vertical": "top",
                                                }
                              }
                              checked={false}
                              desktop={false}
                              disabled={false}
                              focusState="none"
                              insetChildren={false}
                              targetOrigin={
                                        Object {
                                                  "horizontal": "left",
                                                  "vertical": "top",
                                                }
                              }
                    >
                              Dashboard
                    </MenuItem>
          </Link>
          <Link
                    className="nav-link"
                    replace={false}
                    to="/wallets"
          >
                    <MenuItem
                              anchorOrigin={
                                        Object {
                                                  "horizontal": "right",
                                                  "vertical": "top",
                                                }
                              }
                              checked={false}
                              desktop={false}
                              disabled={false}
                              focusState="none"
                              insetChildren={false}
                              targetOrigin={
                                        Object {
                                                  "horizontal": "left",
                                                  "vertical": "top",
                                                }
                              }
                    >
                              Wallets
                    </MenuItem>
          </Link>
          <Link
                    className="nav-link"
                    replace={false}
                    to="/private"
          >
                    <MenuItem
                              anchorOrigin={
                                        Object {
                                                  "horizontal": "right",
                                                  "vertical": "top",
                                                }
                              }
                              checked={false}
                              desktop={false}
                              disabled={false}
                              focusState="none"
                              insetChildren={false}
                              targetOrigin={
                                        Object {
                                                  "horizontal": "left",
                                                  "vertical": "top",
                                                }
                              }
                    >
                              Private
                    </MenuItem>
          </Link>
</Drawer>,
        "className": "drawer-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="logo"
>
              BLOSYM
</div>,
            <Link
              className="nav-link"
              replace={false}
              to="/"
>
              <MenuItem
                            anchorOrigin={
                                          Object {
                                                        "horizontal": "right",
                                                        "vertical": "top",
                                                      }
                            }
                            checked={false}
                            desktop={false}
                            disabled={false}
                            focusState="none"
                            insetChildren={false}
                            targetOrigin={
                                          Object {
                                                        "horizontal": "left",
                                                        "vertical": "top",
                                                      }
                            }
              >
                            Dashboard
              </MenuItem>
</Link>,
            <Link
              className="nav-link"
              replace={false}
              to="/wallets"
>
              <MenuItem
                            anchorOrigin={
                                          Object {
                                                        "horizontal": "right",
                                                        "vertical": "top",
                                                      }
                            }
                            checked={false}
                            desktop={false}
                            disabled={false}
                            focusState="none"
                            insetChildren={false}
                            targetOrigin={
                                          Object {
                                                        "horizontal": "left",
                                                        "vertical": "top",
                                                      }
                            }
              >
                            Wallets
              </MenuItem>
</Link>,
            <Link
              className="nav-link"
              replace={false}
              to="/private"
>
              <MenuItem
                            anchorOrigin={
                                          Object {
                                                        "horizontal": "right",
                                                        "vertical": "top",
                                                      }
                            }
                            checked={false}
                            desktop={false}
                            disabled={false}
                            focusState="none"
                            insetChildren={false}
                            targetOrigin={
                                          Object {
                                                        "horizontal": "left",
                                                        "vertical": "top",
                                                      }
                            }
              >
                            Private
              </MenuItem>
</Link>,
          ],
          "disableSwipeToOpen": false,
          "docked": true,
          "open": true,
          "openSecondary": false,
          "swipeAreaWidth": 30,
          "width": null,
          "zDepth": 2,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BLOSYM",
              "className": "logo",
            },
            "ref": null,
            "rendered": "BLOSYM",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MenuItem
                anchorOrigin={
                                Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                targetOrigin={
                                Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                }
>
                Dashboard
</MenuItem>,
              "className": "nav-link",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "children": "Dashboard",
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
              },
              "ref": null,
              "rendered": "Dashboard",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MenuItem
                anchorOrigin={
                                Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                targetOrigin={
                                Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                }
>
                Wallets
</MenuItem>,
              "className": "nav-link",
              "replace": false,
              "to": "/wallets",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "children": "Wallets",
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
              },
              "ref": null,
              "rendered": "Wallets",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MenuItem
                anchorOrigin={
                                Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                targetOrigin={
                                Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                }
>
                Private
</MenuItem>,
              "className": "nav-link",
              "replace": false,
              "to": "/private",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "children": "Private",
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
              },
              "ref": null,
              "rendered": "Private",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="drawer-container"
>
          <Drawer
                    disableSwipeToOpen={false}
                    docked={true}
                    open={true}
                    openSecondary={false}
                    swipeAreaWidth={30}
                    width={null}
                    zDepth={2}
          >
                    <div
                              className="logo"
                    >
                              BLOSYM
                    </div>
                    <Link
                              className="nav-link"
                              replace={false}
                              to="/"
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        targetOrigin={
                                                  Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                        }
                              >
                                        Dashboard
                              </MenuItem>
                    </Link>
                    <Link
                              className="nav-link"
                              replace={false}
                              to="/wallets"
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        targetOrigin={
                                                  Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                        }
                              >
                                        Wallets
                              </MenuItem>
                    </Link>
                    <Link
                              className="nav-link"
                              replace={false}
                              to="/private"
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        targetOrigin={
                                                  Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                        }
                              >
                                        Private
                              </MenuItem>
                    </Link>
          </Drawer>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Drawer
            disableSwipeToOpen={false}
            docked={true}
            open={true}
            openSecondary={false}
            swipeAreaWidth={30}
            width={null}
            zDepth={2}
>
            <div
                        className="logo"
            >
                        BLOSYM
            </div>
            <Link
                        className="nav-link"
                        replace={false}
                        to="/"
            >
                        <MenuItem
                                    anchorOrigin={
                                                Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                    }
                                    checked={false}
                                    desktop={false}
                                    disabled={false}
                                    focusState="none"
                                    insetChildren={false}
                                    targetOrigin={
                                                Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                    }
                        >
                                    Dashboard
                        </MenuItem>
            </Link>
            <Link
                        className="nav-link"
                        replace={false}
                        to="/wallets"
            >
                        <MenuItem
                                    anchorOrigin={
                                                Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                    }
                                    checked={false}
                                    desktop={false}
                                    disabled={false}
                                    focusState="none"
                                    insetChildren={false}
                                    targetOrigin={
                                                Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                    }
                        >
                                    Wallets
                        </MenuItem>
            </Link>
            <Link
                        className="nav-link"
                        replace={false}
                        to="/private"
            >
                        <MenuItem
                                    anchorOrigin={
                                                Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                    }
                                    checked={false}
                                    desktop={false}
                                    disabled={false}
                                    focusState="none"
                                    insetChildren={false}
                                    targetOrigin={
                                                Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                    }
                        >
                                    Private
                        </MenuItem>
            </Link>
</Drawer>,
          "className": "drawer-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="logo"
>
                BLOSYM
</div>,
              <Link
                className="nav-link"
                replace={false}
                to="/"
>
                <MenuItem
                                anchorOrigin={
                                                Object {
                                                                "horizontal": "right",
                                                                "vertical": "top",
                                                              }
                                }
                                checked={false}
                                desktop={false}
                                disabled={false}
                                focusState="none"
                                insetChildren={false}
                                targetOrigin={
                                                Object {
                                                                "horizontal": "left",
                                                                "vertical": "top",
                                                              }
                                }
                >
                                Dashboard
                </MenuItem>
</Link>,
              <Link
                className="nav-link"
                replace={false}
                to="/wallets"
>
                <MenuItem
                                anchorOrigin={
                                                Object {
                                                                "horizontal": "right",
                                                                "vertical": "top",
                                                              }
                                }
                                checked={false}
                                desktop={false}
                                disabled={false}
                                focusState="none"
                                insetChildren={false}
                                targetOrigin={
                                                Object {
                                                                "horizontal": "left",
                                                                "vertical": "top",
                                                              }
                                }
                >
                                Wallets
                </MenuItem>
</Link>,
              <Link
                className="nav-link"
                replace={false}
                to="/private"
>
                <MenuItem
                                anchorOrigin={
                                                Object {
                                                                "horizontal": "right",
                                                                "vertical": "top",
                                                              }
                                }
                                checked={false}
                                desktop={false}
                                disabled={false}
                                focusState="none"
                                insetChildren={false}
                                targetOrigin={
                                                Object {
                                                                "horizontal": "left",
                                                                "vertical": "top",
                                                              }
                                }
                >
                                Private
                </MenuItem>
</Link>,
            ],
            "disableSwipeToOpen": false,
            "docked": true,
            "open": true,
            "openSecondary": false,
            "swipeAreaWidth": 30,
            "width": null,
            "zDepth": 2,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "BLOSYM",
                "className": "logo",
              },
              "ref": null,
              "rendered": "BLOSYM",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <MenuItem
                  anchorOrigin={
                                    Object {
                                                      "horizontal": "right",
                                                      "vertical": "top",
                                                    }
                  }
                  checked={false}
                  desktop={false}
                  disabled={false}
                  focusState="none"
                  insetChildren={false}
                  targetOrigin={
                                    Object {
                                                      "horizontal": "left",
                                                      "vertical": "top",
                                                    }
                  }
>
                  Dashboard
</MenuItem>,
                "className": "nav-link",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "anchorOrigin": Object {
                    "horizontal": "right",
                    "vertical": "top",
                  },
                  "checked": false,
                  "children": "Dashboard",
                  "desktop": false,
                  "disabled": false,
                  "focusState": "none",
                  "insetChildren": false,
                  "targetOrigin": Object {
                    "horizontal": "left",
                    "vertical": "top",
                  },
                },
                "ref": null,
                "rendered": "Dashboard",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <MenuItem
                  anchorOrigin={
                                    Object {
                                                      "horizontal": "right",
                                                      "vertical": "top",
                                                    }
                  }
                  checked={false}
                  desktop={false}
                  disabled={false}
                  focusState="none"
                  insetChildren={false}
                  targetOrigin={
                                    Object {
                                                      "horizontal": "left",
                                                      "vertical": "top",
                                                    }
                  }
>
                  Wallets
</MenuItem>,
                "className": "nav-link",
                "replace": false,
                "to": "/wallets",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "anchorOrigin": Object {
                    "horizontal": "right",
                    "vertical": "top",
                  },
                  "checked": false,
                  "children": "Wallets",
                  "desktop": false,
                  "disabled": false,
                  "focusState": "none",
                  "insetChildren": false,
                  "targetOrigin": Object {
                    "horizontal": "left",
                    "vertical": "top",
                  },
                },
                "ref": null,
                "rendered": "Wallets",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <MenuItem
                  anchorOrigin={
                                    Object {
                                                      "horizontal": "right",
                                                      "vertical": "top",
                                                    }
                  }
                  checked={false}
                  desktop={false}
                  disabled={false}
                  focusState="none"
                  insetChildren={false}
                  targetOrigin={
                                    Object {
                                                      "horizontal": "left",
                                                      "vertical": "top",
                                                    }
                  }
>
                  Private
</MenuItem>,
                "className": "nav-link",
                "replace": false,
                "to": "/private",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "anchorOrigin": Object {
                    "horizontal": "right",
                    "vertical": "top",
                  },
                  "checked": false,
                  "children": "Private",
                  "desktop": false,
                  "disabled": false,
                  "focusState": "none",
                  "insetChildren": false,
                  "targetOrigin": Object {
                    "horizontal": "left",
                    "vertical": "top",
                  },
                },
                "ref": null,
                "rendered": "Private",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
